### PR TITLE
refactor(trans-button): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by hengkx** [comment]
change log 无能为力了🤣

File: components/_util/transButton.tsx [hunk 1]
```diff
@@ -20,81 +21,63 @@ const inlineStyle: React.CSSProperties = {
...
-    if (autoFocus) {
-      this.focus();
+  React.useEffect(() => {
+    if (props.autoFocus) {
+      if (mergedRef) {
+        mergedRef.current?.focus();
+      }
     }
-  }
+  }, [props.autoFocus]);
```
Comment (zombieJ) [inline components/_util/transButton.tsx:34]: 不用加这个条件，两者不等价。只有 mount 的时候才需要 `autoFocus`

Comment (zombieJ) [reply components/_util/transButton.tsx:34]: 突然想到，这个 `autoFocus` 是不是多余的。实际元素本来就是 focusable 的，应该不需要手工调用。

Comment (hengkx) [reply components/_util/transButton.tsx]: 那我  吧 整个 foucus给移除吧

**Comment by hengkx** [comment]
这个错应该不影响吧

**Pull request merged**